app/src/main/java/org/fossasia/openevent/app/module/event/create/CreateEventFragment.java

```diff
@@ -73,6 +85,81 @@ public void onStart() {
         super.onStart();
         getPresenter().attach(this);
         binding.setEvent(getPresenter().getEvent());
+        getPresenter().start();
+
+        validateUrl(binding.form.ticketUrlLayout);
+        validateUrl(binding.form.logoUrlLayout);
+        validateUrl(binding.form.externalEventUrlLayout);
+        validateUrl(binding.form.originalImageUrlLayout);
+        validateEmail(binding.form.paypalEmailLayout);
+    }
+
+    @Override
+    public void validateUrl(TextInputLayout textInputLayout) {
+        textInputLayout.getEditText().addTextChangedListener(new TextWatcher() {
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                // Nothing here
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                if (ValidateUtils.validateUrl(charSequence.toString())) {
+                    textInputLayout.setError(null);
+                    textInputLayout.setErrorEnabled(false);
+                } else {
+                    textInputLayout.setErrorEnabled(true);
+                    textInputLayout.setError("Please Enter a Valid Url");
+                }
+                if (TextUtils.isEmpty(charSequence)) {
+                    textInputLayout.setError(null);
+                    textInputLayout.setErrorEnabled(false);
+                }
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                // Nothing here
+            }
+        });
+    }
+
+    @Override
+    public void validateEmail(TextInputLayout textInputLayout) {
+        textInputLayout.getEditText().addTextChangedListener(new TextWatcher() {
+
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                // Nothing here
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                if (ValidateUtils.validateEmail(charSequence.toString())) {
+                    textInputLayout.setError(null);
+                    textInputLayout.setErrorEnabled(false);
+                } else {
+                    textInputLayout.setErrorEnabled(true);
+                    textInputLayout.setError("Please Enter a Valid Email");
+                }
+                if (TextUtils.isEmpty(charSequence)) {
+                    textInputLayout.setError(null);
+                    textInputLayout.setErrorEnabled(false);
+                }
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                // Nothing here
+            }
+        });
+    }
```

Comment:
Very nice job! Just last change here, and we are done. These two tasks are essentially same here except two things, validation method and error text. This can be made in a single method which takes a method reference and a string parameter. Also, use strings.xml template to render the text instead

Reply:
Changes Done !!